### ISSUE TITLE
Block Explorer API - fetch events for a transaction

### ISSUE THREAD

**Issue by edd** [vegaprotocol/vega#7251]
# Feature Overview

**In order to** understand what happens with a transaction
**We will** add an API that can fetch the full event stream for a transaction
**So that** the results of a transaction can be shown

# Tasks
As with #7248, I don't really know how much finrmation I'm asking for here or what it looks like. I'm *assuming& that this would end up with something like https://etherscan.io/tx/0x9ebdb8e783f8949e457a5c9f2d741865aca0d3c2eb2b82fedeec6d32809b4bce#statechange (random transaction) where we can show the transfers, or any side effects of a transaction. 

- [ ] Add an endpoint that can return all `bus_events` filtered by `txHash`

# Additional Details
tldr this is an API addition, not a TX change.

This is part of a set of Block Explorer API requests that change the intent of the Block Explorer API from 'return precisely what is in the command' to 'return the details a client needs to render the appropriate details for a transaction'. Ideally we can go through a proper API planning excercise for this, but this ticket can be addressed as is #7248, #7247, #7246, #7245

**Comment by barnabee** [comment]
It would seem like this might want to be a data node API, as IIRC data node is storing tx IDs against events and records in its database, unless we want to create a block explorer bus event store, but if this sort of API is useful then it perhaps ought to be part of data node 🤔

**Comment by edd** [comment]
@barnabee I believe there's a spoken or unspoken limit of how much Block Explorer wants to store about 'chain' data vs 'trading' data. @pscott31 / @jeremyletang can probably debate which is most appropriate with you.

For my part, I don't mind if it's in data node or block explorer apis.

**Comment by pscott31** [comment]
We don't store the events generated by each transaction verbatim, there is some transformation and normalisation that goes on before it gets plonked in the DB so there's no simple table of `txid -> [event]`. We could add that but it would probably ~3-5x the space requirements. We do however store the txid along side every row we put in the database so we could write a query or set of queries to look at each individual table and get the rows that were inserted as a result of a particular transaction.

**Comment by barnabee** [comment]
I'd imagine that those rows would be as good as or bet than bus events in many cases, for instance they'd give you:

- New or changed orders
- Ledger movements (type of these and other data includes a lot of things that might be considered events)
- New trades 
- Proposals and votes
- etc…

There'd definitely be value in looking at some of the bus events for really detailed transaction "logging" (and perhaps this data wouldn't be best placed to be in data node unless it's a specific subset, given the size estimate), but I wonder if the best starting point might be to have a way to query the rows they are already in DN by tx ID?

**Comment by pscott31** [comment]
Yes that (query for x thing by txid) would be very easy to add in and was always something I had planned to do. I guess the only question is is it one API call (all things by txid) or many (orders by txid, proposals by txid etc..)

The only thing that would be awkward to add would be to actually _include_ the txid in the things we return, since we re-use a lot of the core protobuf types which currently don't have a 'txid' field. So you can't currently get a trade back from the API and see what transaction it came from easily (though that is all there in the database)

**Comment by edd** [comment]
The (not that) new `entities` query *sort of* does **some** of this. Instead of leaving this open as a duplicate, I'm closing this ticket and will replace it with tickets for what the entities query failes to do
